engine/src/main/java/org/terasology/rendering/dag/nodes/OpaqueBlocksNode.java

```diff
@@ -98,35 +99,16 @@ public void process() {
             RenderableChunk chunk = renderQueues.chunksOpaque.poll();
 
             if (chunk.hasMesh()) {
+                final ChunkMesh chunkMesh = chunk.getMesh();
                 final Vector3f chunkPosition = chunk.getPosition().toVector3f();
-                final Vector3f chunkPositionRelativeToCamera =
-                        new Vector3f(chunkPosition.x * ChunkConstants.SIZE_X - cameraPosition.x,
-                                chunkPosition.y * ChunkConstants.SIZE_Y - cameraPosition.y,
-                                chunkPosition.z * ChunkConstants.SIZE_Z - cameraPosition.z);
-
-                chunkShader.setFloat3("chunkPositionWorld",
-                        chunkPosition.x * ChunkConstants.SIZE_X,
-                        chunkPosition.y * ChunkConstants.SIZE_Y,
-                        chunkPosition.z * ChunkConstants.SIZE_Z,
-                        true);
-                chunkShader.setFloat("animated", chunk.isAnimated() ? 1.0f : 0.0f, true);
-
-                // Effectively this just positions the chunk appropriately, relative to the camera.
-                // chunkPositionRelativeToCamera = chunkCoordinates * chunkDimensions - cameraCoordinate
-                GL11.glPushMatrix();
-                GL11.glTranslatef(chunkPositionRelativeToCamera.x, chunkPositionRelativeToCamera.y, chunkPositionRelativeToCamera.z);
+
+                chunkMesh.updateMaterial(chunkShader, chunkPosition, chunk.isAnimated());
+                numberOfRenderedTriangles += chunkMesh.render(OPAQUE, chunkPosition, cameraPosition);
 
                 if (renderingDebugConfig.isRenderChunkBoundingBoxes()) {
-                    AABBRenderer aabbRenderer = new AABBRenderer(chunk.getAABB());
```

Comment:
The previous commit introduced a change in the bounding box rendering functionality, confirmed again in the new commit: bounding boxes are rendered only for opaque chunks. They are not rendered in reflections, shadows and for blocks that are not opaque. This is debatable and if you really think I should I could re-introduce them in some of nodes modified by this PR.